evalml/utils/woodwork_utils.py

```diff
@@ -8,68 +8,75 @@
 numeric_and_boolean_ww = [ww.logical_types.Integer, ww.logical_types.Double, ww.logical_types.Boolean]
 
 
+def _numpy_to_pandas(array):
+    if len(array.shape) == 1:
+        data = pd.Series(array)
+    else:
+        data = pd.DataFrame(array)
+    return data
+
+
+def _list_to_pandas(list):
+    return _numpy_to_pandas(np.array(list))
+
+
+_nullable_types = {'Int64', 'Float64', 'boolean'}
+
+
+def _raise_value_error_if_nullable_types_detected(data):
+    types = {data.name: data.dtype} if isinstance(data, pd.Series) else data.dtypes
+    cols_with_nullable_types = {col: str(ptype) for col, ptype in dict(types).items() if str(ptype) in _nullable_types}
+    if cols_with_nullable_types:
+        raise ValueError("Evalml does not support the new pandas nullable types because "
+                         "our dependencies (sklearn, xgboost, lightgbm) do not support them yet."
+                         "If your data does not have missing values, please use the non-nullable types (bool, int64, float64). "
+                         "If your data does have missing values, use float64 for int and float columns and category for boolean columns. "
+                         f"These are the columns with nullable types: {list(cols_with_nullable_types.items())}")
+
+
 def infer_feature_types(data, feature_types=None):
     """Create a Woodwork structure from the given list, pandas, or numpy input, with specified types for columns.
         If a column's type is not specified, it will be inferred by Woodwork.
 
     Arguments:
-        data (pd.DataFrame): Input data to convert to a Woodwork data structure.
+        data (pd.DataFrame, pd.Series): Input data to convert to a Woodwork data structure.
         feature_types (string, ww.logical_type obj, dict, optional): If data is a 2D structure, feature_types must be a dictionary
             mapping column names to the type of data represented in the column. If data is a 1D structure, then feature_types must be
             a Woodwork logical type or a string representing a Woodwork logical type ("Double", "Integer", "Boolean", "Categorical", "Datetime", "NaturalLanguage")
 
     Returns:
         A Woodwork data structure where the data type of each column was either specified or inferred.
     """
-    ww_data = data
-    if isinstance(data, ww.DataTable) or isinstance(data, ww.DataColumn):
-        return ww_data
     if isinstance(data, list):
-        ww_data = np.array(data)
+        data = _list_to_pandas(data)
+    elif isinstance(data, np.ndarray):
+        data = _numpy_to_pandas(data)
 
-    ww_data = ww_data.copy()
-    if len(ww_data.shape) == 1:
-        name = ww_data.name if isinstance(ww_data, pd.Series) else None
-        return ww.DataColumn(ww_data, name=name, logical_type=feature_types)
-    return ww.DataTable(ww_data, logical_types=feature_types)
+    ww_data = data.copy()
 
+    _raise_value_error_if_nullable_types_detected(data)
 
-def _convert_woodwork_types_wrapper(pd_data):
```

Comment:
Don't need convert_woodwork_types_wrapper anymore! I've tested this out on another branch that updates the rest of the codebase to use the latest woodwork.

Reply:
`_convert_woodwork_types_wrapper()`
2021-2021
"Beloved private function. Devoted partner to `infer_feature_types()`. Rest in Power."